### ISSUE TITLE
Whole program optimization

### ISSUE THREAD

**Issue by TobiasKarnat** [open-watcom/open-watcom-v2#180]
Whole program optimization would be a cool new feature. Is that possible?
At least something like gcc -fwhole-program would be nice to have for unity builds.

**Comment by jmalak** [comment]
OW compiles are not able to do any global optimization over more compilation units.

**Comment by TobiasKarnat** [comment]
I know that OW can't do global optimization now, but could this change?

Using unity build (https://en.wikipedia.org/wiki/Single_Compilation_Unit) OW already does some interprocedural optimization, but it is a hack.

Gcc's -fwhole-program does not do optimization between compitlation units, but 
"Assume[s] that the current compilation unit represents the whole program being compiled. All public functions and variables with the exception of main and those merged by attribute externally_visible become static functions and in effect are optimized more aggressively by interprocedural optimizers."

To summarize, the following effects of a new optimization flag for OW would be nice to have:
* Make all public functions and variables with the exception of main and those declared by a new keyword (e.g. __forceextern) become static.
* Remove the copy of a inlined functions which is not declared by the new keyword.
